### PR TITLE
visdiff on windows

### PR DESCRIPTION
**Pull request by chromakode**
:eyeglasses: @keybase/react-hackers 

Alright! This is finally running on Jenkins, and ready to take a look at.

I've moved the visdiff runner into its own directory and separated out the dependencies. I anticipate it's going to grow a bit in the future when we add HTML output for diffing, so it seems like the right time to break it out. Importantly, this also makes it convenient to install as a binary outside of the `desktop` source tree, so that it doesn't get blown away when commits are checked out (which caused cryptic errors from the `s3` module).

The visdiff package is transpiled via a npm prepublish hook. Setting it up takes a few steps:

```sh
# from root repo dir
cd visdiff
npm install  # we need to install babel build deps in the visdiff package first
cd ..
npm install ./visdiff  # transpile and install a binary in root node_modules
cd desktop
npm install
../node_modules/.bin/keybase-visdiff
```

I tried some other variations, but didn't come up with a simpler approach. I didn't want to put `keybase-visdiff` in `devDependencies` of desktop because it's primarily used in CI and pulls in its own unrelated deps. Long-term, the package should get built and vendored into our JS deps like every other package, at which point we'll stop having to do this dance.

It was necessary to hack in the commit ids for demonstrating this branch, because `render-screenshots` is run from the state of the commit that is being checked, and it doesn't work on windows until this this changeset is applied.

### Outstanding tasks:
- [x] Finalize Windows Jenkins setup where this will live
- [x] Update Linux-JS Jenkins setup to build and use new setup, remove s3cmd cruft
- [x] [idea] Resolve HEAD^...HEAD style definitions from `argv` with `git rev-parse` so the runner can take arguments again

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 090fe2fcfd72...1101810a6022 introduced some visual changes:
 * Changed: **Folders_Clear_History_Confirmation-Normal_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466731915952-090fe2fcfd72-1101810a6022-win32/Folders_Clear_History_Confirmation-Normal_Public-090fe2fcfd72.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466731915952-090fe2fcfd72-1101810a6022-win32/Folders_Clear_History_Confirmation-Normal_Public-1101810a6022.png)  
   Folders_Clear_History_Confirmation-Normal_Public comparison
 * Changed: **Folders_Clear_History_Confirmation-Normal_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466731915952-090fe2fcfd72-1101810a6022-win32/Folders_Clear_History_Confirmation-Normal_Private-090fe2fcfd72.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466731915952-090fe2fcfd72-1101810a6022-win32/Folders_Clear_History_Confirmation-Normal_Private-1101810a6022.png)  
   Folders_Clear_History_Confirmation-Normal_Private comparison
 * Changed: **Folders_Clear_History_Confirmation-Long_Name_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466731915952-090fe2fcfd72-1101810a6022-win32/Folders_Clear_History_Confirmation-Long_Name_Public-090fe2fcfd72.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466731915952-090fe2fcfd72-1101810a6022-win32/Folders_Clear_History_Confirmation-Long_Name_Public-1101810a6022.png)  
   Folders_Clear_History_Confirmation-Long_Name_Public comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 808bca765658...4c2ba835117b introduced some visual changes on win32:
 * Changed: **Folders_Clear_History_Confirmation-Normal_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466732125871-808bca765658-4c2ba835117b-win32/Folders_Clear_History_Confirmation-Normal_Public-808bca765658.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466732125871-808bca765658-4c2ba835117b-win32/Folders_Clear_History_Confirmation-Normal_Public-4c2ba835117b.png)  
   Folders_Clear_History_Confirmation-Normal_Public comparison
 * Changed: **Folders_Clear_History_Confirmation-Normal_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466732125871-808bca765658-4c2ba835117b-win32/Folders_Clear_History_Confirmation-Normal_Private-808bca765658.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466732125871-808bca765658-4c2ba835117b-win32/Folders_Clear_History_Confirmation-Normal_Private-4c2ba835117b.png)  
   Folders_Clear_History_Confirmation-Normal_Private comparison
 * Changed: **Folders_Clear_History_Confirmation-Long_Name_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466732125871-808bca765658-4c2ba835117b-win32/Folders_Clear_History_Confirmation-Long_Name_Public-808bca765658.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466732125871-808bca765658-4c2ba835117b-win32/Folders_Clear_History_Confirmation-Long_Name_Public-4c2ba835117b.png)  
   Folders_Clear_History_Confirmation-Long_Name_Public comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 40191f81f485...38adfa269cb0 introduced some visual changes on win32:
 * Changed: **Folders_Clear_History_Confirmation-Normal_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466732554805-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Normal_Public-40191f81f485.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466732554805-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Normal_Public-38adfa269cb0.png)  
   Folders_Clear_History_Confirmation-Normal_Public comparison
 * Changed: **Folders_Clear_History_Confirmation-Normal_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466732554805-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Normal_Private-40191f81f485.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466732554805-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Normal_Private-38adfa269cb0.png)  
   Folders_Clear_History_Confirmation-Normal_Private comparison
 * Changed: **Folders_Clear_History_Confirmation-Long_Name_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466732554805-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Long_Name_Public-40191f81f485.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466732554805-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Long_Name_Public-38adfa269cb0.png)  
   Folders_Clear_History_Confirmation-Long_Name_Public comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 40191f81f485...38adfa269cb0 introduced some visual changes on win32:
 * Changed: **Folders_Clear_History_Confirmation-Normal_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466735064046-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Normal_Public-40191f81f485.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466735064046-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Normal_Public-38adfa269cb0.png)  
   Folders_Clear_History_Confirmation-Normal_Public comparison
 * Changed: **Folders_Clear_History_Confirmation-Normal_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466735064046-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Normal_Private-40191f81f485.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466735064046-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Normal_Private-38adfa269cb0.png)  
   Folders_Clear_History_Confirmation-Normal_Private comparison
 * Changed: **Folders_Clear_History_Confirmation-Long_Name_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466735064046-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Long_Name_Public-40191f81f485.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466735064046-40191f81f485-38adfa269cb0-win32/Folders_Clear_History_Confirmation-Long_Name_Public-38adfa269cb0.png)  
   Folders_Clear_History_Confirmation-Long_Name_Public comparison

File: visdiff/src/index.js [hunk 1]
```diff
@@ -193,16 +224,12 @@ function processDiff (commitRange, results) {
...
-if (process.argv.length !== 3) {
-  console.log(`Usage: node ${path.basename(process.argv[1])} COMMIT1..COMMIT2`)
-  process.exit(1)
-}
-
-const commitRange = process.argv[2]
-  .split(/\.{2,3}/)  // Travis gives us ranges like START...END
-  .map(s => s.substr(0, 12))  // trim the hashes a bit for shorter paths
+const commitRange = [
+  '40191f81f485483254311178b1a7d22af5567bfd'.trim(),
```
Comment (chrisnojima) [inline visdiff/src/index.js]: 🔥

File: visdiff/src/index.js [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 import _ from 'lodash'
+import mkdirp from 'mkdirp'
+import del from 'del'
 import gm from 'gm'
 import github from 'octonode'
+import s3 from 's3'
 
-const BUCKET_S3 = 's3://keybase-app-visdiff'
+const BUCKET_S3 = 'keybase-app-visdiff'
 const BUCKET_HTTP = 'http://keybase-app-visdiff.s3.amazonaws.com'
```
Comment (gabriel) [inline visdiff/src/index.js]: https everywhere?

you can use the bucket syntax: 'https://s3.amazonaws.com/keybase-app-visdiff/' if node has cert wildcart problems with s3

Comment (chromakode) [inline visdiff/src/index.js]: This is just for formatting the comments on GitHub; the s3 upload goes through the `s3` module which presumably uses SSL. Switching these URLs to be HTTPS wouldn't hurt though, good idea!

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 583b8975ba39...976a7f93010f introduced some visual changes on win32:
 * Changed: **Folders_Clear_History_Confirmation-Normal_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466800294418-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Public-583b8975ba39.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466800294418-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Public-976a7f93010f.png)  
   Folders_Clear_History_Confirmation-Normal_Public comparison
 * Changed: **Folders_Clear_History_Confirmation-Normal_Private** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466800294418-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Private-583b8975ba39.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466800294418-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Private-976a7f93010f.png)  
   Folders_Clear_History_Confirmation-Normal_Private comparison
 * Changed: **Folders_Clear_History_Confirmation-Long_Name_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466800294418-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Long_Name_Public-583b8975ba39.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466800294418-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Long_Name_Public-976a7f93010f.png)  
   Folders_Clear_History_Confirmation-Long_Name_Public comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 583b8975ba39...976a7f93010f introduced some visual changes on win32:
 * Changed: **Folders_Clear_History_Confirmation-Normal_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466800670739-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Public-583b8975ba39.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466800670739-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Public-976a7f93010f.png)  
   Folders_Clear_History_Confirmation-Normal_Public comparison
 * Changed: **Folders_Clear_History_Confirmation-Normal_Private** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466800670739-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Private-583b8975ba39.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466800670739-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Private-976a7f93010f.png)  
   Folders_Clear_History_Confirmation-Normal_Private comparison
 * Changed: **Folders_Clear_History_Confirmation-Long_Name_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466800670739-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Long_Name_Public-583b8975ba39.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466800670739-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Long_Name_Public-976a7f93010f.png)  
   Folders_Clear_History_Confirmation-Long_Name_Public comparison

**Comment by chromakode** [comment]
@keybase/react-hackers updated and ready for another look!

**Comment by chromakode** [comment]
retest this please

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 583b8975ba39...976a7f93010f introduced some visual changes on win32:
 * Changed: **Folders_Clear_History_Confirmation-Normal_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466807541931-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Public-583b8975ba39.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466807541931-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Public-976a7f93010f.png)  
   Folders_Clear_History_Confirmation-Normal_Public comparison
 * Changed: **Folders_Clear_History_Confirmation-Normal_Private** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466807541931-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Private-583b8975ba39.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466807541931-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Normal_Private-976a7f93010f.png)  
   Folders_Clear_History_Confirmation-Normal_Private comparison
 * Changed: **Folders_Clear_History_Confirmation-Long_Name_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466807541931-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Long_Name_Public-583b8975ba39.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466807541931-583b8975ba39-976a7f93010f-win32/Folders_Clear_History_Confirmation-Long_Name_Public-976a7f93010f.png)  
   Folders_Clear_History_Confirmation-Long_Name_Public comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 38a912081b9c...490a0cb687ed introduced some visual changes on win32:
 * Changed: **Folders_Clear_History_Confirmation-Normal_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466809140711-38a912081b9c-490a0cb687ed-win32/Folders_Clear_History_Confirmation-Normal_Public-38a912081b9c.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466809140711-38a912081b9c-490a0cb687ed-win32/Folders_Clear_History_Confirmation-Normal_Public-490a0cb687ed.png)  
   Folders_Clear_History_Confirmation-Normal_Public comparison
 * Changed: **Folders_Clear_History_Confirmation-Normal_Private** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466809140711-38a912081b9c-490a0cb687ed-win32/Folders_Clear_History_Confirmation-Normal_Private-38a912081b9c.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466809140711-38a912081b9c-490a0cb687ed-win32/Folders_Clear_History_Confirmation-Normal_Private-490a0cb687ed.png)  
   Folders_Clear_History_Confirmation-Normal_Private comparison
 * Changed: **Folders_Clear_History_Confirmation-Long_Name_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466809140711-38a912081b9c-490a0cb687ed-win32/Folders_Clear_History_Confirmation-Long_Name_Public-38a912081b9c.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466809140711-38a912081b9c-490a0cb687ed-win32/Folders_Clear_History_Confirmation-Long_Name_Public-490a0cb687ed.png)  
   Folders_Clear_History_Confirmation-Long_Name_Public comparison

**Comment by MarcoPolo** [comment]
👍

**Comment by chromakode** [comment]
retest this please

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 4a1a2fe11ce8...66f49bdb023d introduced some visual changes on win32:
 * Changed: **Folders_Clear_History_Confirmation-Normal_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466811238223-4a1a2fe11ce8-66f49bdb023d-win32/Folders_Clear_History_Confirmation-Normal_Public-4a1a2fe11ce8.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466811238223-4a1a2fe11ce8-66f49bdb023d-win32/Folders_Clear_History_Confirmation-Normal_Public-66f49bdb023d.png)  
   Folders_Clear_History_Confirmation-Normal_Public comparison
 * Changed: **Folders_Clear_History_Confirmation-Normal_Private** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466811238223-4a1a2fe11ce8-66f49bdb023d-win32/Folders_Clear_History_Confirmation-Normal_Private-4a1a2fe11ce8.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466811238223-4a1a2fe11ce8-66f49bdb023d-win32/Folders_Clear_History_Confirmation-Normal_Private-66f49bdb023d.png)  
   Folders_Clear_History_Confirmation-Normal_Private comparison
 * Changed: **Folders_Clear_History_Confirmation-Long_Name_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466811238223-4a1a2fe11ce8-66f49bdb023d-win32/Folders_Clear_History_Confirmation-Long_Name_Public-4a1a2fe11ce8.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466811238223-4a1a2fe11ce8-66f49bdb023d-win32/Folders_Clear_History_Confirmation-Long_Name_Public-66f49bdb023d.png)  
   Folders_Clear_History_Confirmation-Long_Name_Public comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b61db7ef9844...30f46d4a3d96 introduced some visual changes on linux:
 * Changed: **Folders_Clear_History_Confirmation-Normal_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466811352024-b61db7ef9844-30f46d4a3d96-linux/Folders_Clear_History_Confirmation-Normal_Public-b61db7ef9844.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466811352024-b61db7ef9844-30f46d4a3d96-linux/Folders_Clear_History_Confirmation-Normal_Public-30f46d4a3d96.png)  
   Folders_Clear_History_Confirmation-Normal_Public comparison
 * Changed: **Folders_Clear_History_Confirmation-Normal_Private** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466811352024-b61db7ef9844-30f46d4a3d96-linux/Folders_Clear_History_Confirmation-Normal_Private-b61db7ef9844.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466811352024-b61db7ef9844-30f46d4a3d96-linux/Folders_Clear_History_Confirmation-Normal_Private-30f46d4a3d96.png)  
   Folders_Clear_History_Confirmation-Normal_Private comparison
 * Changed: **Folders_Clear_History_Confirmation-Long_Name_Public** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1466811352024-b61db7ef9844-30f46d4a3d96-linux/Folders_Clear_History_Confirmation-Long_Name_Public-b61db7ef9844.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1466811352024-b61db7ef9844-30f46d4a3d96-linux/Folders_Clear_History_Confirmation-Long_Name_Public-30f46d4a3d96.png)  
   Folders_Clear_History_Confirmation-Long_Name_Public comparison

**Pull request merged**